xbmc/interfaces/python/XBPython.cpp

```diff
@@ -50,6 +50,12 @@ XBPython::~XBPython()
 {
   XBMC_TRACE;
   CServiceBroker::GetAnnouncementManager()->RemoveAnnouncer(this);
+
+  if (Py_IsInitialized())
+  {
+    PyThreadState_Swap(PyInterpreterState_ThreadHead(PyInterpreterState_Main()));
```

Comment:
PyInterpreterState_Main() requires Python >= 3.7. Is that dependency acceptable?

Test build on openSUSE Leap 15.4 with Python 3.6 is failing.

Reply:
You are right this function was already implemented (PythonInvoker.cpp) but with a version check. I will update the code.

Reply:
Thanks, is compiling and starting (no further tests yet).